### PR TITLE
Remove `eofs`

### PR DESCRIPTION
**Pull request by Zeitsperre**
### Pull Request Checklist:
- [x] This PR addresses an already opened issue (for bug fixes / features)
    - This PR addresses problems raised in #1620 
- [ ] Tests for the changes have been added (for bug fixes / features)
  - [ ] (If applicable) Documentation has been added / updated (for bug fixes / features)
- [x] CHANGES.rst has been updated (with summary of main changes)
  - [x] Link to issue (:issue:`number`) and pull request (:pull:`number`) has been added

### What kind of change does this PR introduce?

* The `eofs` library has been removed from the optional dependencies.
* The empirical orthogonal functionality (`xclim.sdba.properties.first_eof` and associated test) has been removed from the code base.
* CI builds no longer test `eofs` functionality.

### Does this PR introduce a breaking change?

Yes. `eofs` has been completely removed since this violated its GPLv3 license. Associated functionality is reproducible by users 

### Other information:

Should we be offering users a note about how to migrate? I feel like it's not necessary.

### CODE REVIEW

**Review by aulemahal — APPROVED** [review]
What does @juliettelavoie think of this.
Were you using that property anywhere ? If yes, we'll need to reimplement it in a private script somewhere.

**Comment by juliettelavoie** [comment]
I don't have active scripts that use that property. I am okay to delete it without reimplementing it until we have a clear need for it.

File: CHANGES.rst [hunk 1]
```diff
@@ -24,7 +24,7 @@ Breaking changes
 * `bump2version` has been replaced with `bump-my-version` to bump the version number using configurations set in the `pyproject.toml` file. (:issue:`1557`, :pull:`1569`).
 * `xclim`'s units registry and units formatting are now extended from `cf-xarray`. The exponent sign "^" is now never added in the ``units`` attribute. For example, square meters are given as "m2" instead of "m^2" by xclim, both are still accepted as input. (:issue:`1010`, :pull:`1590`).
 * `yamale` is now listed as a core dependency (was previously listed in the `dev` installation recipe). (:issue:`1595`, :pull:`1596`).
-* Due to a licensing limitation, the bias correction algorithm based on `eofs` (``xclim.sdba.properties.first_eof``)  has been removed from `xclim`. (:issue:`1620`).
+* Due to a licensing limitation, the experimental bias correction algorithm based on `eofs` (``xclim.sdba.properties.first_eof``) has been removed from `xclim`. (:issue:`1620`, :pull:`1621`).
```
Comment (juliettelavoie) [inline CHANGES.rst:27]: not a big deal but it is not really a bias correction alg, more a tool to evaluate different bc methods.

Comment (Zeitsperre) [reply CHANGES.rst:27]: Good to know, thanks! I have no idea what goes on in `sdba`. I'll update this.

Comment (huard) [reply CHANGES.rst:27]:
```suggestion
* Due to a licensing limitation, the calculation of empirical orthogonal function  based on `eofs` (``xclim.sdba.properties.first_eof``) has been removed from `xclim`. (:issue:`1620`, :pull:`1621`).
```

**Review by huard — APPROVED** [review]

**Pull request merged**